### PR TITLE
Add test for host registration using capsule cname

### PR DESCRIPTION
**Pull request by shweta83**
Add test for host registration using capsule cname.
BZ#2158959

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7712
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname --external-logging
Test Result : ========== 16 deselected, 69 warnings, 2 errors in 1361.83s (0:22:41) ==========
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8448
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname --external-logging
Test Result : ===== 2 failed, 18 deselected, 77 warnings, 2 errors in 1449.10s (0:24:09) =====
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8677
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname --external-logging
Test Result : ========== 2 passed, 18 deselected, 88 warnings in 1550.69s (0:25:50) ==========
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py -k test_positive_register_host_with_capsule_cname

File: tests/foreman/api/test_registration.py [hunk 1]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
...
+        activation_keys=[module_activation_key.name],
+        location=module_location,
+        force=True,
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    rhel_contenthost.execute(
+        f'echo "{module_capsule_configured.ip} {module_capsule_configured.hostname} {capsule_cname}'
+        f'" >> /etc/hosts'
+    )
+
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:498]: isn't there any direct way from smart_proxy to handle setting the cname's?

File: tests/foreman/api/test_registration.py [hunk 2]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
...
+    :expectedresults: Host registers successfully with capsule cname.
+
+    :BZ: 2158959
+
+    :customerscenario: true
+
+    :parametrized: yes
+    """
+    capsule_cname = "capsule.example.com"
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:478]:
```suggestion
    capsule_cname = 'capsule.example.com'
```

File: tests/foreman/api/test_registration.py [hunk 3]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
     # check if the certificate file is refreshed
     ca_file_after_refresh = len(str(rhel_contenthost.execute(f'cat {katello_ca_crt_path}')))
     assert ca_cert_file == ca_file_after_refresh
+
+
+@pytest.mark.rhel_ver_match('[^7]')
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:451]: is it really required to run for all rhel versions? I feel default should be suffice to test

File: tests/foreman/api/test_registration.py [hunk 4]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
...
+def test_positive_register_host_with_capsule_cname(
+    module_sca_manifest_org,
+    module_location,
+    module_target_sat,
+    rhel_contenthost,
+    module_activation_key,
+    module_capsule_configured,
+):
+    """Verify host registration works with proxy and proxy_cname
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:460]: we're just testing with proxy_cname here, right?

File: tests/foreman/api/test_registration.py [hunk 5]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
...
+
+    :id: 86b5c329-7784-43f4-bb67-67ce5a24ce17
+
+    :steps:
+        1. Get a configure capsule registered with Satellite.
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:465]:
```suggestion
        1. Configure external capsule which is registered to Satellite
```

File: tests/foreman/api/test_registration.py [hunk 6]
```diff
@@ -445,3 +446,66 @@ def test_positive_katello_ca_crt_refresh(
...
+        2. Verify Host registers to Capsule successfully.
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:466]: Host or Capsule?

**Pull request closed without merging**